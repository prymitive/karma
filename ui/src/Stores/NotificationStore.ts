import { action, computed, observable, makeObservable } from "mobx";

export interface NotificationT {
  id: string;
  type: "warning" | "error";
  title: string;
  message: string;
  timestamp: Date;
  dismissed: boolean;
  autoDismissed: boolean;
  source: "alertmanager" | "system";
  sourceId?: string; // For alertmanager notifications, this would be the upstream name
  occurrenceCount: number; // Track how many times this notification has occurred
}

class NotificationStore {
  notifications: NotificationT[] = [];

  constructor() {
    makeObservable(this, {
      notifications: observable,
      addNotification: action,
      dismissNotification: action,
      dismissAllBySource: action,
      clearDismissed: action,
      activeNotifications: computed,
      dismissedNotifications: computed,
      warningCount: computed,
      errorCount: computed,
      totalActiveCount: computed,
    });
  }

  addNotification = (
    notification: Omit<
      NotificationT,
      "id" | "timestamp" | "dismissed" | "autoDismissed" | "occurrenceCount"
    >,
  ) => {
    // Check if we already have an active notification for this source
    const existingActiveIndex = this.notifications.findIndex(
      (n) =>
        n.sourceId === notification.sourceId &&
        n.source === notification.source &&
        !n.dismissed,
    );

    // Check if we have a previously dismissed notification for this source
    const existingDismissedIndex = this.notifications.findIndex(
      (n) =>
        n.sourceId === notification.sourceId &&
        n.source === notification.source &&
        n.dismissed,
    );

    if (existingActiveIndex >= 0) {
      // Update existing active notification
      const newNotification: NotificationT = {
        ...notification,
        id: this.notifications[existingActiveIndex].id,
        timestamp: new Date(),
        dismissed: false,
        autoDismissed: false,
        occurrenceCount:
          this.notifications[existingActiveIndex].occurrenceCount + 1,
      };
      this.notifications[existingActiveIndex] = newNotification;
    } else if (existingDismissedIndex >= 0) {
      // Instead of creating a new notification, increment the count on the dismissed one
      const dismissedNotification = this.notifications[existingDismissedIndex];
      dismissedNotification.occurrenceCount += 1;
      dismissedNotification.timestamp = new Date(); // Update timestamp to reflect latest occurrence
      // Note: We keep it dismissed but show the updated count in history
    } else {
      // Add new notification
      const newNotification: NotificationT = {
        ...notification,
        id: `${Date.now()}-${Math.random().toString(36).substr(2, 9)}`,
        timestamp: new Date(),
        dismissed: false,
        autoDismissed: false,
        occurrenceCount: 1,
      };
      this.notifications.push(newNotification);
    }

    // Keep only last 50 notifications to prevent memory issues
    if (this.notifications.length > 50) {
      this.notifications = this.notifications.slice(-50);
    }
  };

  dismissNotification = (id: string, isAutoDismiss = false) => {
    const notification = this.notifications.find((n) => n.id === id);
    if (notification) {
      notification.dismissed = true;
      notification.autoDismissed = isAutoDismiss;
    }
  };

  dismissAllBySource = (
    source: string,
    sourceId?: string,
    isAutoDismiss = false,
  ) => {
    this.notifications.forEach((notification) => {
      if (
        notification.source === source &&
        (!sourceId || notification.sourceId === sourceId) &&
        !notification.dismissed
      ) {
        notification.dismissed = true;
        notification.autoDismissed = isAutoDismiss;
      }
    });
  };

  clearDismissed = () => {
    this.notifications = this.notifications.filter((n) => !n.dismissed);
  };

  get activeNotifications(): NotificationT[] {
    return this.notifications.filter((n) => !n.dismissed);
  }

  get dismissedNotifications(): NotificationT[] {
    return this.notifications.filter((n) => n.dismissed).slice(-20); // Keep last 20 dismissed
  }

  get warningCount(): number {
    return this.activeNotifications.filter((n) => n.type === "warning").length;
  }

  get errorCount(): number {
    return this.activeNotifications.filter((n) => n.type === "error").length;
  }

  get totalActiveCount(): number {
    return this.activeNotifications.length;
  }
}

export { NotificationStore };
export const notificationStore = new NotificationStore();
