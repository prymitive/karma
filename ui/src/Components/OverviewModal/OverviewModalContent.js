import React, { useState } from "react";
import PropTypes from "prop-types";

import { observer } from "mobx-react-lite";

import { AlertStore } from "Stores/AlertStore";
import { TooltipWrapper } from "Components/TooltipWrapper";
import { LabelWithPercent } from "Components/Labels/LabelWithPercent";
import { ToggleIcon } from "Components/ToggleIcon";

const TableRows = observer(({ alertStore, nameStats }) =>
  nameStats.map((nameStats) => (
    <tr key={nameStats.name}>
      <td width="25%" className="text-nowrap mw-100 p-1">
        <span className="badge badge-light components-label mx-0 mt-0 mb-auto pl-0 text-left">
          <span className="bg-primary text-white mr-1 px-1 components-labelWithPercent-percent">
            {nameStats.hits}
          </span>
          {nameStats.name}
        </span>
      </td>
      <td width="75%" className="mw-100 p-1">
        {nameStats.values.slice(0, 9).map((valueStats, i) => (
          <LabelWithPercent
            alertStore={alertStore}
            key={valueStats.value}
            name={nameStats.name}
            value={valueStats.value}
            hits={valueStats.hits}
            percent={valueStats.percent}
            offset={valueStats.offset}
            isActive={
              alertStore.filters.values.filter((f) => f.raw === valueStats.raw)
                .length > 0
            }
          />
        ))}
        {nameStats.values.length > 9 ? (
          <div className="components-label badge my-2 text-muted mw-100">
            +{nameStats.values.length - 9} more
          </div>
        ) : null}
      </td>
    </tr>
  ))
);

const LabelsTable = observer(
  ({ alertStore, showAllLabels, toggleAllLabels }) => (
    <React.Fragment>
      <table
        className="table table-borderless top-labels"
        style={{ tableLayout: "fixed" }}
      >
        <tbody className="mw-100">
          <TableRows
            alertStore={alertStore}
            nameStats={alertStore.data.counters.filter(
              (nameStats) => nameStats.hits >= alertStore.info.totalAlerts
            )}
          ></TableRows>
          {alertStore.data.counters.filter(
            (nameStats) => nameStats.hits < alertStore.info.totalAlerts
          ).length > 0 ? (
            <tr>
              <td colSpan="2" className="px-1 py-0">
                <TooltipWrapper
                  title="Toggle all / only common labels"
                  delay={[3000, 100]}
                >
                  <ToggleIcon
                    isOpen={showAllLabels}
                    className="cursor-pointer text-muted"
                    onClick={toggleAllLabels}
                  />
                </TooltipWrapper>
              </td>
            </tr>
          ) : null}
          {showAllLabels ? (
            <TableRows
              alertStore={alertStore}
              nameStats={alertStore.data.counters.filter(
                (nameStats) => nameStats.hits < alertStore.info.totalAlerts
              )}
            ></TableRows>
          ) : null}
        </tbody>
      </table>
    </React.Fragment>
  )
);

const NothingToShow = () => (
  <div className="jumbotron bg-transparent">
    <h1 className="display-5 text-placeholder text-center">
      No labels to display
    </h1>
  </div>
);

const OverviewModalContent = observer(({ alertStore, onHide }) => {
  const [showAllLabels, setShowAllLabels] = useState(false);
  return (
    <React.Fragment>
      <div className="modal-header">
        <h5 className="modal-title">Overview</h5>
        <button type="button" className="close" onClick={onHide}>
          <span className="align-middle">&times;</span>
        </button>
      </div>
      <div className="modal-body">
        {alertStore.data.counters.length === 0 ? (
          <NothingToShow />
        ) : (
          <LabelsTable
            alertStore={alertStore}
            showAllLabels={showAllLabels}
            toggleAllLabels={() => setShowAllLabels(!showAllLabels)}
          />
        )}
      </div>
    </React.Fragment>
  );
});
OverviewModalContent.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  onHide: PropTypes.func.isRequired,
};

export { OverviewModalContent };
