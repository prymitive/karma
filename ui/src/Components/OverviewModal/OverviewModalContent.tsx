import { FC, useState } from "react";

import { observer } from "mobx-react-lite";

import { APILabelCounterT } from "Models/APITypes";
import { AlertStore } from "Stores/AlertStore";
import { TooltipWrapper } from "Components/TooltipWrapper";
import LabelWithPercent from "Components/Labels/LabelWithPercent";
import { ToggleIcon } from "Components/ToggleIcon";

const TableRows: FC<{
  alertStore: AlertStore;
  nameStats: APILabelCounterT[];
}> = observer(({ alertStore, nameStats }) => (
  <>
    {nameStats.map((nameStats) => (
      <tr key={nameStats.name}>
        <td width="25%" className="text-nowrap mw-100 p-1">
          <span className="badge bg-light components-label mx-0 mt-0 mb-auto ps-0 text-start">
            <span className="bg-primary text-white me-1 px-1 components-labelWithPercent-percent">
              {nameStats.hits}
            </span>
            {nameStats.name}
          </span>
        </td>
        <td width="75%" className="mw-100 p-1">
          {nameStats.values.slice(0, 9).map((valueStats) => (
            <LabelWithPercent
              alertStore={alertStore}
              key={valueStats.value}
              name={nameStats.name}
              value={valueStats.value}
              hits={valueStats.hits}
              percent={valueStats.percent}
              offset={valueStats.offset}
              isActive={
                alertStore.filters.values.filter(
                  (f) => f.raw === valueStats.raw
                ).length > 0
              }
            />
          ))}
          {nameStats.values.length > 9 ? (
            <div className="components-label badge my-2 text-muted mw-100">
              +{nameStats.values.length - 9} more
            </div>
          ) : null}
        </td>
      </tr>
    ))}
  </>
));

const LabelsTable: FC<{
  alertStore: AlertStore;
  showAllLabels: boolean;
  toggleAllLabels: () => void;
}> = observer(({ alertStore, showAllLabels, toggleAllLabels }) => (
  <>
    <table
      className="table table-borderless top-labels"
      style={{ tableLayout: "fixed" }}
    >
      <tbody className="mw-100">
        <TableRows
          alertStore={alertStore}
          nameStats={alertStore.data.counters.filter(
            (nameStats) => nameStats.hits >= alertStore.info.totalAlerts
          )}
        ></TableRows>
        {alertStore.data.counters.filter(
          (nameStats) => nameStats.hits < alertStore.info.totalAlerts
        ).length > 0 ? (
          <tr>
            <td colSpan={2} className="px-1 py-0">
              <TooltipWrapper title="Toggle all / only common labels">
                <span
                  className="badge components-label cursor-pointer with-click"
                  onClick={toggleAllLabels}
                >
                  <ToggleIcon isOpen={showAllLabels} className="text-muted" />
                </span>
              </TooltipWrapper>
            </td>
          </tr>
        ) : null}
        {showAllLabels ? (
          <TableRows
            alertStore={alertStore}
            nameStats={alertStore.data.counters.filter(
              (nameStats) => nameStats.hits < alertStore.info.totalAlerts
            )}
          ></TableRows>
        ) : null}
      </tbody>
    </table>
  </>
));

const NothingToShow: FC = () => (
  <div className="px-2 py-5 bg-transparent">
    <h1 className="display-5 text-placeholder text-center">
      No labels to display
    </h1>
  </div>
);

const OverviewModalContent: FC<{
  alertStore: AlertStore;
  onHide: () => void;
}> = observer(({ alertStore, onHide }) => {
  const [showAllLabels, setShowAllLabels] = useState<boolean>(false);
  return (
    <>
      <div className="modal-header">
        <h5 className="modal-title">Overview</h5>
        <button type="button" className="btn-close" onClick={onHide}></button>
      </div>
      <div className="modal-body">
        {alertStore.data.counters.length === 0 ? (
          <NothingToShow />
        ) : (
          <LabelsTable
            alertStore={alertStore}
            showAllLabels={showAllLabels}
            toggleAllLabels={() => setShowAllLabels(!showAllLabels)}
          />
        )}
      </div>
    </>
  );
});

export { OverviewModalContent };
