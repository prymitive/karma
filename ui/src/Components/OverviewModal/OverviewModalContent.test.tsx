import { mount } from "enzyme";

import toDiffableHtml from "diffable-html";

import { AlertStore, NewUnappliedFilter } from "Stores/AlertStore";
import { OverviewModalContent } from "./OverviewModalContent";

let alertStore: AlertStore;
const onHide = jest.fn();

beforeEach(() => {
  alertStore = new AlertStore([]);
  onHide.mockClear();
});

afterEach(() => {
  jest.restoreAllMocks();
});

const MountedOverviewModalContent = () =>
  // we have multiple fragments and enzyme only renders the first one
  // in html() and text(), debug() would work but it's noisy
  // https://github.com/airbnb/enzyme/issues/1213
  mount(
    <span>
      <OverviewModalContent alertStore={alertStore} onHide={onHide} />
    </span>
  );

describe("<OverviewModalContent />", () => {
  it("matches snapshot with labels to show", () => {
    alertStore.filters.setFilterValues([
      NewUnappliedFilter("abc=xyz"),
      NewUnappliedFilter("foo=bar"),
    ]);
    alertStore.data.setCounters([
      {
        name: "foo",
        hits: 16,
        values: [
          { value: "bar1", raw: "foo=bar1", hits: 8, percent: 50, offset: 0 },
          { value: "bar2", raw: "foo=bar2", hits: 4, percent: 25, offset: 50 },
          { value: "bar3", raw: "foo=bar3", hits: 4, percent: 25, offset: 75 },
        ],
      },
      {
        name: "bar",
        hits: 20,
        values: Array.from(Array(20).keys()).map((i) => ({
          value: `baz${i + 1}`,
          raw: `bar=baz${i + 1}`,
          hits: 1,
          percent: 5,
          offset: i * 5,
        })),
      },
      {
        name: "alertname",
        hits: 5,
        values: [
          {
            value: "Host_Down",
            raw: "alertname=Host_Down",
            hits: 5,
            percent: 100,
            offset: 0,
          },
        ],
      },
    ]);

    const tree = MountedOverviewModalContent();
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with no labels to show", () => {
    alertStore.data.setCounters([]);
    const tree = MountedOverviewModalContent();
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("renders all labels after expand button click", () => {
    alertStore.info.setTotalAlerts(5);
    alertStore.data.setCounters([
      {
        name: "foo",
        hits: 5,
        values: [
          { value: "bar", raw: "foo=bar", hits: 5, percent: 100, offset: 0 },
        ],
      },
      {
        name: "bar",
        hits: 3,
        values: [
          {
            value: "foo",
            raw: "bar=foo",
            hits: 3,
            percent: 100,
            offset: 0,
          },
        ],
      },
    ]);
    const tree = MountedOverviewModalContent();

    expect(tree.find("span.components-label")).toHaveLength(2 + 1); // +1 for toggle icon
    expect(tree.find("span.components-label").at(0).text()).toBe("5foo");
    expect(tree.find("span.components-label").at(1).text()).toBe("5foo: bar");

    tree.find("span.badge.cursor-pointer.with-click").simulate("click");

    expect(tree.find("span.components-label")).toHaveLength(4 + 1); // +1 for toggle icon
    expect(tree.find("span.components-label").at(3).text()).toBe("3bar");
    expect(tree.find("span.components-label").at(4).text()).toBe("3bar: foo");
  });
});
