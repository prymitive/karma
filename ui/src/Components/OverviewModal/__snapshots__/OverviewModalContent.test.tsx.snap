// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverviewModalContent /> matches snapshot with labels to show 1`] = `
"
<span>
  <div class=\\"modal-header\\">
    <h5 class=\\"modal-title\\">
      Overview
    </h5>
    <button type=\\"button\\"
            class=\\"btn-close\\"
    >
    </button>
  </div>
  <div class=\\"modal-body\\">
    <table class=\\"table table-borderless top-labels\\"
           style=\\"table-layout: fixed;\\"
    >
      <tbody class=\\"mw-100\\">
        <tr>
          <td width=\\"25%\\"
              class=\\"text-nowrap mw-100 p-1\\"
          >
            <span class=\\"badge bg-light components-label mx-0 mt-0 mb-auto ps-0 text-start\\">
              <span class=\\"bg-primary text-white me-1 px-1 components-labelWithPercent-percent\\">
                16
              </span>
              foo
            </span>
          </td>
          <td width=\\"75%\\"
              class=\\"mw-100 p-1\\"
          >
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  8
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    foo:
                  </span>
                  <span class=\\"components-label-value\\">
                    bar1
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-warning\\"
                     role=\\"progressbar\\"
                     style=\\"width: 50%;\\"
                     aria-valuenow=\\"50\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  4
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    foo:
                  </span>
                  <span class=\\"components-label-value\\">
                    bar2
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 50%;\\"
                     aria-valuenow=\\"50\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 25%;\\"
                     aria-valuenow=\\"25\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  4
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    foo:
                  </span>
                  <span class=\\"components-label-value\\">
                    bar3
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 75%;\\"
                     aria-valuenow=\\"75\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 25%;\\"
                     aria-valuenow=\\"25\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td width=\\"25%\\"
              class=\\"text-nowrap mw-100 p-1\\"
          >
            <span class=\\"badge bg-light components-label mx-0 mt-0 mb-auto ps-0 text-start\\">
              <span class=\\"bg-primary text-white me-1 px-1 components-labelWithPercent-percent\\">
                20
              </span>
              bar
            </span>
          </td>
          <td width=\\"75%\\"
              class=\\"mw-100 p-1\\"
          >
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz1
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz2
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz3
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 10%;\\"
                     aria-valuenow=\\"10\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz4
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 15%;\\"
                     aria-valuenow=\\"15\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz5
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 20%;\\"
                     aria-valuenow=\\"20\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz6
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 25%;\\"
                     aria-valuenow=\\"25\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz7
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 30%;\\"
                     aria-valuenow=\\"30\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz8
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 35%;\\"
                     aria-valuenow=\\"35\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-default components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  1
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    bar:
                  </span>
                  <span class=\\"components-label-value\\">
                    baz9
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-transparent\\"
                     role=\\"progressbar\\"
                     style=\\"width: 40%;\\"
                     aria-valuenow=\\"40\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
                <div class=\\"progress-bar bg-success\\"
                     role=\\"progressbar\\"
                     style=\\"width: 5%;\\"
                     aria-valuenow=\\"5\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
            <div class=\\"components-label badge my-2 text-muted mw-100\\">
              +11 more
            </div>
          </td>
        </tr>
        <tr>
          <td width=\\"25%\\"
              class=\\"text-nowrap mw-100 p-1\\"
          >
            <span class=\\"badge bg-light components-label mx-0 mt-0 mb-auto ps-0 text-start\\">
              <span class=\\"bg-primary text-white me-1 px-1 components-labelWithPercent-percent\\">
                5
              </span>
              alertname
            </span>
          </td>
          <td width=\\"75%\\"
              class=\\"mw-100 p-1\\"
          >
            <div class=\\"d-inline-block mw-100\\">
              <span class=\\"components-label badge bg-dark components-label-dark components-label-with-hover mb-0 ps-0 text-start\\">
                <span class=\\"me-1 px-1 bg-primary text-white components-labelWithPercent-percent\\">
                  5
                </span>
                <span>
                  <span class=\\"components-label-name\\">
                    alertname:
                  </span>
                  <span class=\\"components-label-value\\">
                    Host_Down
                  </span>
                </span>
              </span>
              <div class=\\"progress components-labelWithPercent-progress me-1\\">
                <div class=\\"progress-bar bg-danger\\"
                     role=\\"progressbar\\"
                     style=\\"width: 100%;\\"
                     aria-valuenow=\\"100\\"
                     aria-valuemin=\\"0\\"
                     aria-valuemax=\\"100\\"
                >
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</span>
"
`;

exports[`<OverviewModalContent /> matches snapshot with no labels to show 1`] = `
"
<span>
  <div class=\\"modal-header\\">
    <h5 class=\\"modal-title\\">
      Overview
    </h5>
    <button type=\\"button\\"
            class=\\"btn-close\\"
    >
    </button>
  </div>
  <div class=\\"modal-body\\">
    <div class=\\"px-2 py-5 bg-transparent\\">
      <h1 class=\\"display-5 text-placeholder text-center\\">
        No labels to display
      </h1>
    </div>
  </div>
</span>
"
`;
